### PR TITLE
Miner Loot Shuffle MK III

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
Might as well call it a rebalance at this point ^_^
## About The Pull Request

Moves god's eyes to fallen angel loot table thus killing the angel might be more worth wile to do...
Removes spell blade and fire wand from drake/bubble gum table
Replaces fire wand with meat hook
Replaces spell blade with holoparasight
Moves the holoparasight, meat hook and eye of god from spike loot table.
Makes each type of spike and boss have its own type of chest
Their 3 types, magic, armor/weapon, and misc!
These three new types pick almost all the old loot but into different pools making the ones that people like much more clear to get, such as getting lets state ship in a bottle, now you have more odds to get it if you kill the correct spike type!

This helps combat just flat out mean Rng were you can fight tooth and nail for a golith spike chest and get... A worthless ship in a bottle you already can make with just some bones and now the countless hides you have from killing the spike...

Adds 2 more crates to Arena Shuttle as well as 4-4-4 of each new type of chest

Added 4 magic ~~wands~~ Spell books! These little guys are like really cool and fire pages of things! Reloadable too with adding in of pages
1 Book heals all main types of damage by 5
1 Book just deals damage
1 Book shocks and tries to be like a really really bad taser bolt that is weak and useless*
1 Book Sets people on fire like a bullet

Adds in 3 healing potions
The red one heals "100" damage of the main types
One heals the mind clearly most stats affects and tramma
Well one heals organ damage! Yay!

New pick that is as supper dooper good! Drills 3 tiles and at even faster speeds as the diamond one this replaces the diamond one.

 Replaces the katana with a non-cursed weaker one. Maybe people will like this one better?

## Why It's Good For The Game

Makes fighting lava land megafuna more worth wile and less of a Well if get lucky with spikes I can just never deal with that and more stable rounds without the issue of 4 holo parasights or other things of that sort just via spike drops that are kinda easy to cheese and such.
This also takes out the frankly overly good fire wand aka 1 hit killer
and the also grate spell blade that nuggets people has 50% block iirc and is amazing.

Spike/Miniboss changes -
These changes to make each type of boss/spike have its own loot table makes you have a more informed, if you look into it, odds of what to get as well as helps people know/target what kinda mobs they want to fight againts well not really buffing anything other then what tendrils to kill first :P

For bosses these are a more rng and ontop of Rng if you fight them, you flat out dont know the miniboss you get, sadly I have no plans to make each miniboss have a tell before you spawn it.

I do really really really think these are good change to make. As they are not that nerfing or buffing, just shifting around power to more informed things so that as a player you can strategist better!

Current pix can be gotton via the vender or other means rather lamely

Current katana is both shockingly hated and powerfull meaning that people that do use it are able to really easy take down targets well everyone more so is like `This is trash` and leaves it always do to bone spear being "better" as is the crusher

## Changelog
:cl: Putnam for debugging ammo loading! Thx!!
del: Removed spell blade, diamond picaxe, and fire wand from lava land loot tables
tweak: Each mini boss now has its own type of crate.
tweak: Each spike has its own type of crate that it pulls its now smaller loot table from
balance: Moved god eye from spike loot table to hard spawn collosses fight
balance: Moved holoparasight from spike loot table to bubble gum
balance: Moved magic meat hook from spike loot table to drake
add: 2 more crates to Arena Shuttle as well as 4-4-4 of each new type of chest
balance: Replaced the diamond pick loot with a better one
tweak: Replaced the cursted katana with a non cursted verson that deals half the damage and has less block!
add: Three new potions, blue heals the mind like a mama potion, Green heals the soul aka the organs! Lastly Red heals the body, by 100 damage of each main types. Best not to waste them!
add: Four more "wands" Spellbooks! These fun little guys shoot out its own pages to do the affect, one will set the target on fire like a bullet, one will harm them a bit, one will heal the target a small bit - How nice! Last one will give them a few statis affects like a taser bolt but without as much power or tasing affect
/:cl:

### _**Check out pr #12353 and pr #12431  to see commits that dislike this pr.**_

### CODE REVIEW

**Comment by Trilbyspaceclone** [comment]
image

Kevs smile of approvement on the PR

**Comment by YakumoChen** [comment]
Nice items, but maybe add them without the shuffling

File: code/modules/mining/equipment/mining_tools.dm [hunk 1]
```diff
@@ -66,6 +66,16 @@
 	force = 19
 	custom_materials = list(/datum/material/diamond=4000)
 
+/obj/item/pickaxe/rosegold
+	name = "rose gold pickaxe"
+	icon_state = "rgpickaxe"
+	item_state = "rgpickaxe"
+	toolspeed = 0.1
+	desc = "A pickaxe with a light rose gold head and some red glowing ruins. Extremely robust at cracking rock walls and digging up dirt."
```
Comment (ghost) [inline code/modules/mining/equipment/mining_tools.dm]: runes

File: code/modules/projectiles/ammunition/special/magic.dm [hunk 2]
```diff
@@ -43,3 +43,17 @@
 
 /obj/item/ammo_casing/magic/locker
 	projectile_type = /obj/item/projectile/magic/locker
+
+//Spell book ammo casing
+/obj/item/ammo_casing/magic/book/book
```
Comment (Ghommie) [inline code/modules/projectiles/ammunition/special/magic.dm]:
```suggestion
/obj/item/ammo_casing/magic/book
```

File: code/modules/projectiles/projectile/magic/spellcard.dm [hunk 3]
```diff
@@ -4,3 +4,60 @@
...
+	damage = 4
+	var/fire_stacks = 4
+
+/obj/item/projectile/magic/spellcard/book/spark/on_hit(atom/target, blocked = FALSE)
+	. = ..()
+	var/mob/living/carbon/M = target
+	if(iscarbon(target))
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			return BULLET_ACT_BLOCK
```
Comment (Ghommie) [inline code/modules/projectiles/projectile/magic/spellcard.dm:27]: Move the anti magic check before the iscarbon one.

File: code/modules/projectiles/projectile/magic/spellcard.dm [hunk 4]
```diff
@@ -4,3 +4,60 @@
...
+		else
+			M.adjust_fire_stacks(fire_stacks)
+			M.IgniteMob()
+			return
+	else
+		damage = 20 //If we are a simplemob we deal 5x damage
```
Comment (Ghommie) [inline code/modules/projectiles/projectile/magic/spellcard.dm:34]: This won't do anything, bullet damage is applied to the mob before `on_hit` is called.

File: code/modules/projectiles/projectile/magic/spellcard.dm [hunk 5]
```diff
@@ -4,3 +4,60 @@
...
+	jitter = 20
+	knockdown = 10
+
+/obj/item/projectile/magic/spellcard/book/heal
+	damage = 0
+
+/obj/item/projectile/magic/spellcard/book/heal/on_hit(atom/target, blocked = FALSE)
+	. = ..()
+	var/mob/living/carbon/M = target
+	if(iscarbon(target))
```
Comment (Ghommie) [inline code/modules/projectiles/projectile/magic/spellcard.dm]: Just early return if the target is not a living mob. `anti_magic_check` is a `mob` proc, while most health adjustment procs are on `mob/living`.

File: code/modules/projectiles/projectile/magic/spellcard.dm [hunk 6]
```diff
@@ -4,3 +4,60 @@
...
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			return BULLET_ACT_BLOCK
+		else
```
Comment (Ghommie) [inline code/modules/projectiles/projectile/magic/spellcard.dm]: Useless indentation, remove.

File: code/modules/projectiles/projectile/magic/spellcard.dm [hunk 7]
```diff
@@ -4,3 +4,60 @@
...
+			M.adjustBruteLoss(-5) //HEALS
+			M.adjustOxyLoss(-5)
+			M.adjustBruteLoss(-5)
+			M.adjustFireLoss(-5)
+			M.adjustToxLoss(-5, TRUE) //heals TOXINLOVERs
+			M.adjustCloneLoss(-5)
+			M.adjustStaminaLoss(-5)
+			return
+	else
+		return
```
Comment (Ghommie) [inline code/modules/projectiles/projectile/magic/spellcard.dm]: unnecessary, remove.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 8]
```diff
@@ -2195,6 +2195,58 @@
 	color = "#f7685e"
 	metabolization_rate = REAGENTS_METABOLISM * 0.25
 
+/datum/reagent/red_ichor
```
Comment (Ghommie) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: These three chemicals are some serious OP filler content. I can't quite agree on them.

File: code/modules/projectiles/guns/magic/spell_book.dm [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+		return
+	if(target == user)
+		to_chat(user, "The book has [charges] pages\s remaining.</span>")
+	else
+		. = ..()
+	update_icon()
+
+/obj/item/gun/magic/wand/book/attackby(obj/item/S, mob/living/user, params)
+	if(!istype(S, /obj/item/paper))
+		return
```
Comment (Ghommie) [inline code/modules/projectiles/guns/magic/spell_book.dm]:
```suggestion
		return ..()
```

File: code/modules/projectiles/guns/magic/spell_book.dm [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+	if(charges < max_charges)
+		charges++
+		recharge_newshot()
+		to_chat(user, "You add a new page to [src].</span>")
+		qdel(S)
+		update_icon()
+		process()
+	else
+		to_chat(user, "The [src] has no more room for pages!</span>")
+		return
```
Comment (Ghommie) [inline code/modules/projectiles/guns/magic/spell_book.dm]: unnecessary return, the proc already ends here.

File: code/modules/projectiles/guns/magic/spell_book.dm [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+	icon_state = "book"
+	w_class = WEIGHT_CLASS_NORMAL
+	charges = 10 //We start with max pages
+	max_charges = 10
+	variable_charges = FALSE
+	self_attack = FALSE
+
+/obj/item/gun/magic/wand/book/examine(mob/user)
+	. = ..()
+	. += "Has [charges] pages\s remaining."
```
Comment (Ghommie) [inline code/modules/projectiles/guns/magic/spell_book.dm]: `/obj/item/gun/magic/wand/examine()` already tells the amount of charges left. Just make the noun in that string a variable if you wish it to say "page\s" instead of "charge\s" and remove this copypasta.

File: code/modules/projectiles/guns/magic/spell_book.dm [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+
+/obj/item/gun/magic/wand/book/afterattack(atom/target, mob/living/user)
+	if(charges == 0)
+		shoot_with_empty_chamber(user)
+		return
+	if(target == user)
+		to_chat(user, "The book has [charges] pages\s remaining.</span>")
+	else
+		. = ..()
+	update_icon()
```
Comment (Ghommie) [inline code/modules/projectiles/guns/magic/spell_book.dm]:
```suggestion
/obj/item/gun/magic/wand/book/zap_self(mob/living/user)
	to_chat(user, "The book has [charges] pages\s remaining.</span>")
```

Bad copypasta. The only thing that differed from parent's was the `target == user` interaction.

File: code/modules/projectiles/guns/magic/wand.dm [hunk 13]
```diff
@@ -8,6 +8,7 @@
 	can_charge = 0
 	max_charges = 100 //100, 50, 50, 34 (max charge distribution by 25%ths)
 	var/variable_charges = 1
+	var/self_attack = TRUE
```
Comment (Ghommie) [inline code/modules/projectiles/guns/magic/wand.dm]: This variable is not needed anymore.

File: code/modules/projectiles/guns/magic/spell_book.dm [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+	name = "blank spellbook"
+	desc = "It's not just a book, it's a SPELL book!"
+	ammo_type = /obj/item/ammo_casing/magic
+	icon = 'icons/obj/library.dmi'
+	icon_state = "book"
+	w_class = WEIGHT_CLASS_NORMAL
+	charges = 10 //We start with max pages
+	max_charges = 10
+	variable_charges = FALSE
+	self_attack = FALSE
```
Comment (Ghommie) [inline code/modules/projectiles/guns/magic/spell_book.dm]:
```suggestion
	no_den_usage = FALSE
```

File: code/modules/projectiles/guns/magic/spell_book.dm [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+
+/obj/item/gun/magic/wand/book/examine(mob/user)
+	. = ..()
+	. += "Has [charges] pages\s remaining."
+
+/obj/item/gun/magic/wand/book/attack(atom/target, mob/living/user)
+	if(target == user)
+		return
+	..()
```
Comment (Ghommie) [inline code/modules/projectiles/guns/magic/spell_book.dm]: Copypasta from `wand.dm`, remove.

File: code/modules/projectiles/projectile/magic/spellcard.dm [hunk 16]
```diff
@@ -4,3 +4,60 @@
...
+
+/obj/item/projectile/magic/spellcard/book/shock
+	damage = 0
+	stamina = 11
+	stutter = 5
+	jitter = 20
+	knockdown = 10
+
+/obj/item/projectile/magic/spellcard/book/heal
+	damage = 0
```
Comment (Ghommie) [inline code/modules/projectiles/projectile/magic/spellcard.dm:44]:
```suggestion
	damage = 0
	nodamage  = TRUE
```

File: code/modules/projectiles/projectile/magic/spellcard.dm [hunk 17]
```diff
@@ -4,3 +4,60 @@
 	icon_state = "spellcard"
 	damage_type = BURN
 	damage = 2
+
+/obj/item/projectile/magic/spellcard/book
+	nodamage = FALSE
```
Comment (Ghommie) [inline code/modules/projectiles/projectile/magic/spellcard.dm:9]: Already false, no need to specify it again.

File: code/modules/projectiles/projectile/magic/spellcard.dm [hunk 18]
```diff
@@ -4,3 +4,60 @@
...
+	name = "enchanted page"
+	desc = "A piece of paper enchanted to give it extreme durability and stiffness, along with a very hot burn to anyone unfortunate enough to get hit by a charged one."
+	icon_state = "spellcard"
+	damage_type = BURN
```
Comment (Ghommie) [inline code/modules/projectiles/projectile/magic/spellcard.dm:13]: same as parent, remove.

File: code/modules/projectiles/projectile/magic/spellcard.dm [hunk 19]
```diff
@@ -4,3 +4,60 @@
...
+	damage = 12
+	flag = "magic"
+
+/obj/item/projectile/magic/spellcard/book/spark
+	damage = 4
+	var/fire_stacks = 4
```
Comment (Ghommie) [inline code/modules/projectiles/projectile/magic/spellcard.dm:19]: no need for this variable, just use the number.

**Review by Ghommie — CHANGES_REQUESTED** [review]
I can't help but feel you are being obstinate with moving some of the better tendril loot to the bosses.

I like spell books (though there should be a minor do_after delay for reloading, paper is plentily available), I don't like the ichor and pickaxe. I'm fine with loot being split between the three types.

**Comment by YakumoChen** [comment]
I'm pretty heavily against splitting them still, it feels arbitrary and it divides the loot further into tiers, the magic tendril has almost all the best things for example. I'm betting it also fucks with legion drops, etc.

File: code/modules/projectiles/projectile/magic/spellcard.dm (hunk 4 above)
Comment (Trilbyspaceclone) [reply code/modules/projectiles/projectile/magic/spellcard.dm:34]: In my personal testing this works... Somehow... Im really not sure on it but it did deal 20 damage to the simple mobs I tested it on

File: code/modules/projectiles/guns/magic/spell_book.dm (hunk 14 above)
Comment (Trilbyspaceclone) [reply code/modules/projectiles/guns/magic/spell_book.dm]: No.

File: code/modules/projectiles/projectile/magic/spellcard.dm (hunk 17 above)
Comment (Trilbyspaceclone) [reply code/modules/projectiles/projectile/magic/spellcard.dm:9]: No its not

File: code/modules/projectiles/projectile/magic/spellcard.dm (hunk 18 above)
Comment (Trilbyspaceclone) [reply code/modules/projectiles/projectile/magic/spellcard.dm:13]: No its not

File: code/modules/projectiles/projectile/magic/spellcard.dm (hunk 19 above)
Comment (Trilbyspaceclone) [reply code/modules/projectiles/projectile/magic/spellcard.dm:19]: This var is not defined  tho

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 8 above)
Comment (Trilbyspaceclone) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: Its healing potions!
They cant be synthed cant be multy at all
Red heals about 100~ of each damage
Green heals internals all the way up - Good for freshly defibbed miners or the like
Blue heals stam/mind

File: code/modules/projectiles/guns/magic/spell_book.dm (hunk 11 above)
Comment (Trilbyspaceclone) [reply code/modules/projectiles/guns/magic/spell_book.dm]: What?

File: code/modules/projectiles/guns/magic/spell_book.dm (hunk 10 above)
Comment (Trilbyspaceclone) [reply code/modules/projectiles/guns/magic/spell_book.dm]: Safty Return. I use them just in case

File: code/modules/projectiles/projectile/magic/spellcard.dm (hunk 7 above)
Comment (Trilbyspaceclone) [reply code/modules/projectiles/projectile/magic/spellcard.dm]: Is wanted

File: code/modules/projectiles/projectile/magic/spellcard.dm (hunk 5 above)
Comment (Trilbyspaceclone) [reply code/modules/projectiles/projectile/magic/spellcard.dm]: what?

File: code/modules/projectiles/projectile/magic/spellcard.dm (hunk 17 above)
Comment (Trilbyspaceclone) [reply code/modules/projectiles/projectile/magic/spellcard.dm:9]: ``/obj/item/projectile/magic

	name = "bolt of nothing"

	icon_state = "energy"

	damage = 0

	damage_type = OXY

	nodamage = 1

	armour_penetration = 100

	flag = "magic" ``

I dont want it to be TRUE I want that to be FALSE

File: code/modules/projectiles/projectile/magic/spellcard.dm (hunk 5 above)
Comment (Trilbyspaceclone) [reply code/modules/projectiles/projectile/magic/spellcard.dm]: I do want this to work on the dead...

**Comment by Ghommie** [comment]
Gotta testmerge for a community feedback. I'm a bit skeptic about the loot split up as Chen pointed out.

**Comment by ghost** [comment]
community feedback:

why are we buffing mining? did bobjoga put you up to this? please do not code content for skyrat up on citadel to avoid feature freeze it's just kind of a dick move

actual feedback:
 why do we have literally all of the best fucking loot on watcher tendrils, all of the worst loot on legion tendrils, and only 3 decent items out of 11 items on the goliath tendrils
 why are we straight buffing megafauna loot drops by adding extra op gamer gear onto them??
 what the fuck are we thinking
 this is a real skyrat moment

to elaborate:
 watcher tendrils have the best current mining loot. Cursed heart, holoparasite, rod of asclepius, summon item, memento mori, and the two new op healing items to replace aheal legion cores getting nerfed.

elaboration on why these items are good since people think they're all bad for some reason:
 cursed heart is equivalent to a free legion core every 2-4~ seconds on demand, as well as immunity to bloodloss and therefore pseudo-immunity to sharp weapon wounds
 rod of asclepius is an aoe heal and can give you a free 40 healing every 2 ticks while having a guardian that only takes 5% damage from all sources. Pacifism is a minor downside if you have a stand and/or friend because they can attack for you and take no damage.
 summon item is just good. I don't need to elaborate on this.
 memento mori makes you pseudo immortal. Only works in pvp because lavaland deaths gib. Combine with rod of asclepius to basically never die. Also if you glue memento mori it gets quite quirky.
 heal wand is ok. Should be in its own pr so it gets merged because this pr isn't going to be merged.
 potions are powercreep but funny meme. I'm susceptible to meme pr's but it's bad balancing even if it's funny.

goliath tendrils only decent drops are the new 3 books.
 All of the hardsuit drops are bad because they give you slowdown. Useless on lavaland and only useful in edge cases on station.
 Nullrod drops are only useful on dynamic. Wizard rounds are too short for these to be actually relevant in a normal round.
 Mod disks are bad. Enough said.
 Inferno grenade is ok but it's just kinda shitty pvp plasmafire item. Steal a plasma can and you have a better version of this. Used to be somewhat useful when you could plasmafire lava, as it could let you basalt an entire river.
 New katana is just a shittier crusher. Why would you use this? Crusher has 100% armor penetration and also works on lavaland...


legion tendrils have literally nothing fucking good holy shit these are the tendrils you want to kill least so you're giving them the worst loot??
 voodoo, babel, skateboard, pickaxe, and ship in a bottle aren't useful they're just funny shitpost items
 mod disks are just bad as previously stated
 wisp lantern is kinda ok but colossus drops eye of god guarantee now so wisp lantern is worthless
 malf upgrade is quirky powercreep but is only a 50% chance so whatever


tl;dr: if you haven't read the code for this pr yet, it just removes the rng in getting good lavaland items and guarantees you'll get op items when you go mining. No mergies senseless miner buff pr from someone who doesn't understand gamebalance pls.

**Comment by YakumoChen** [comment]
My very first comment in the first closed PR about moving all the top tier items to megafauna being a fucking horrible idea still applies

**Comment by silicons** [comment]
the new loot are alright from what i've seen (maybe a bit spammable but ehh there's worse)

splitting up and reducing rng is bad for reasons already outlined by yakumo/nameko. sometimes you get good loot sometimes you get bad loot sometimes you get an average roll. making it so killing something ensures you get a roll from one of a few items is terrible.

**Comment by SmArtKar** [comment]
Fix the travis? Also, mining doesnt need buffs at all.

**Comment by Ludox235** [comment]
Don't merge this, only reason miners aren't that op (which they are :) ) is because the loot is just diluted between shit and super good.

Making the loot table less diluted will just mean more gamer Aheals meathook katana memes.

**Comment by Trilbyspaceclone** [comment]
@Ludox235 please make sure to understand the pr before commiting, if you were to read it and understand the changes you would have relised the katana is nerfed
Meat hooks are no longer teneral dropped
And that theirs no aheals in them

File: code/game/objects/items/weaponry.dm [hunk 20]
```diff
@@ -237,6 +237,11 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 	resistance_flags = FIRE_PROOF
 	total_mass = TOTAL_MASS_MEDIEVAL_WEAPON
 
+/obj/item/katana/lavaland
+	desc = "Woefully underpowered in Lavaland."
```
Comment (ghost) [inline code/game/objects/items/weaponry.dm:241]: the true design of mining: all of the mining gear SUCKS on mining and is OP on station

Comment (Trilbyspaceclone) [reply code/game/objects/items/weaponry.dm:241]: I thought it was a good joke tbh, its p-good on station imo

**Comment by AristoVagrant** [comment]
Honestly, this doesn't really look like it'll solve any problems with mining, and instead looks like it'll create it's own host of problems.

File: code/game/objects/items/weaponry.dm (hunk 20 above)
Comment (ghost) [reply code/game/objects/items/weaponry.dm:241]: yes that's why the description is funny

**Review by silicons — CHANGES_REQUESTED** [review]
no rng removal and do not give extra loot for megafauna kills unless said megafauna doesn't have their own unique loot, please.

**Comment by Trilbyspaceclone** [comment]
Sure

**Comment by YakumoChen** [comment]
The megafauna changes still aren't undone

The rng changes are still there, its just lazily patched over instead of being completely reverted.

**Comment by Trilbyspaceclone** [comment]
@YakumoChen Megafauna changes are done wdym
And the RnG changes for tenderals are reverted as asked

**Comment by YakumoChen** [comment]
I feel like you didn't read Kevin's request at all

**Review by silicons — CHANGES_REQUESTED** [review]
removing rng drop is unacceptable. replacing fireball wand in the drop that has it is fine, adding new interesting loot is fine, potentially removing some extremely useless loot in exchange is fine, but making things more static vs rng is not okay.
